Provider crashing when using "headers" on config

Hi,
when i trying to use the "headers" argument in provider configuration, the provider crashes.

Here is my configuration:

```
terraform {
  required_version = ">=v1.0.6"
  required_providers {
  [...]
    argocd = {
      source  = "oboukili/argocd"
      version = "~> 3.0.1"
    }
  }
}

provider "argocd" {
  server_addr = "${var.argocd_ui_dns_name}.${var.cloudflare_zones["tech"].domain}:443"
  username    = "admin"
  password    = base64decode(data.kubernetes_resource.argocd_password.object.data.password)
  headers = [
    "CF-Access-Client-Id :${data.aws_ssm_parameter.argocd_cloudflare_access_service_token_client_id.value}",
    "CF-Access-Client-Secret: ${data.aws_ssm_parameter.argocd_cloudflare_access_service_token_client_secret.value}"
  ]
}

resource "argocd_project" "core" {
  metadata {
    name      = "core"
    namespace = "argocd"
    labels = {
      component = "core"
    }
  }

  spec {
    description  = "Project that contains all apps needed to run other apps"
    source_repos = ["*"]

    destination {
      server    = "in-cluster"
      namespace = "core"
    }
  }
}
```

Here is the log:

```
argocd_project.core[0]: Creating...
╷
│ Error: Plugin did not respond
│ 
│   with argocd_project.core[0],
│   on argocd.tf line 79, in resource "argocd_project" "core":
│   79: resource "argocd_project" "core" {
│ 
│ The plugin encountered an error, and failed to respond to the plugin.(*GRPCProvider).ApplyResourceChange call. The plugin logs may contain more details.
╵

Stack trace from the terraform-provider-argocd_v3.0.1 plugin:

panic: interface conversion: interface {} is *schema.Set, not []string

goroutine 67 [running]:
github.com/oboukili/terraform-provider-argocd/argocd.initApiClient(0x14000db8800, 0x0, 0x0, 0x0, 0x0)
	github.com/oboukili/terraform-provider-argocd/argocd/provider.go:221 +0x1338
github.com/oboukili/terraform-provider-argocd/argocd.(*ServerInterface).initClients(0x1400018c840, 0x0, 0x0)
	github.com/oboukili/terraform-provider-argocd/argocd/features.go:67 +0x79c
github.com/oboukili/terraform-provider-argocd/argocd.resourceArgoCDProjectCreate(0x1071f7898, 0x1400018d3e0, 0x14000db8e80, 0x106f24540, 0x1400018c840, 0x14000aac910, 0x14000ba5100, 0x10512a300)
	github.com/oboukili/terraform-provider-argocd/argocd/resource_argocd_project.go:58 +0x2e4
github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema.(*Resource).create(0x140003bbea0, 0x1071f7828, 0x14000b05b80, 0x14000db8e80, 0x106f24540, 0x1400018c840, 0x0, 0x0, 0x0)
	github.com/hashicorp/terraform-plugin-sdk/v2@v2.7.1/helper/schema/resource.go:330 +0x118
github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema.(*Resource).Apply(0x140003bbea0, 0x1071f7828, 0x14000b05b80, 0x140002287e0, 0x14000ba5100, 0x106f24540, 0x1400018c840, 0x0, 0x0, 0x0, ...)
	github.com/hashicorp/terraform-plugin-sdk/v2@v2.7.1/helper/schema/resource.go:456 +0x4ec
github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema.(*GRPCProviderServer).ApplyResourceChange(0x1400000c3f0, 0x1071f7828, 0x14000b05b80, 0x1400064fbd0, 0x14000b05b80, 0x107049a40, 0x0)
	github.com/hashicorp/terraform-plugin-sdk/v2@v2.7.1/helper/schema/grpc_provider.go:955 +0x6f8
github.com/hashicorp/terraform-plugin-go/tfprotov5/server.(*server).ApplyResourceChange(0x140001bf0e0, 0x1071f78d0, 0x14000b05b80, 0x14000119490, 0x140001bf0e0, 0x1053e14f0, 0x106fce240)
	github.com/hashicorp/terraform-plugin-go@v0.3.0/tfprotov5/server/server.go:332 +0x94
github.com/hashicorp/terraform-plugin-go/tfprotov5/internal/tfplugin5._Provider_ApplyResourceChange_Handler(0x107049a40, 0x140001bf0e0, 0x1071f78d0, 0x14000b21bf0, 0x14000785920, 0x0, 0x1071f78d0, 0x14000b21bf0, 0x140001ccc00, 0x3a2)
	github.com/hashicorp/terraform-plugin-go@v0.3.0/tfprotov5/internal/tfplugin5/tfplugin5_grpc.pb.go:380 +0x1c8
google.golang.org/grpc.(*Server).processUnaryRPC(0x1400047f180, 0x107211e98, 0x14000583800, 0x14000b25d40, 0x140005924b0, 0x10839f180, 0x0, 0x0, 0x0)
	google.golang.org/grpc@v1.40.0/server.go:1297 +0x3e8
google.golang.org/grpc.(*Server).handleStream(0x1400047f180, 0x107211e98, 0x14000583800, 0x14000b25d40, 0x0)
	google.golang.org/grpc@v1.40.0/server.go:1626 +0xa50
google.golang.org/grpc.(*Server).serveStreams.func1.2(0x1400092c160, 0x1400047f180, 0x107211e98, 0x14000583800, 0x14000b25d40)
	google.golang.org/grpc@v1.40.0/server.go:941 +0x94
created by google.golang.org/grpc.(*Server).serveStreams.func1
	google.golang.org/grpc@v1.40.0/server.go:939 +0x1f8

Error: The terraform-provider-argocd_v3.0.1 plugin crashed!

This is always indicative of a bug within the plugin. It would be immensely
helpful if you could report the crash with the plugin's maintainers so that it
can be fixed. The output above should help diagnose the issue.
```

Thank you in advance

--- a/argocd/provider.go
+++ b/argocd/provider.go
@@ -221,7 +221,12 @@ func initApiClient(d *schema.ResourceData) (
 		opts.PortForwardNamespace = v.(string)
 	}
 	if v, ok := d.GetOk("headers"); ok {
-		opts.Headers = v.([]string)
+		_headers := v.(*schema.Set).List()
+		var headers = make([]string, len(_headers))
+		for i, _header := range _headers {
+			headers[i] = _header.(string)
+		}
+		opts.Headers = headers
 	}
 	if _, ok := d.GetOk("kubernetes"); ok {
 		opts.KubeOverrides = &clientcmd.ConfigOverrides{}

--- a/argocd/resource_argocd_application_test.go
+++ b/argocd/resource_argocd_application_test.go
@@ -618,6 +618,25 @@ func TestAccArgoCDApplication_Info(t *testing.T) {
 		}})
 }
 
+func TestProvider_headers(t *testing.T) {
+	name := acctest.RandomWithPrefix("test-acc")
+	resource.Test(t, resource.TestCase{
+		PreCheck:          func() { testAccPreCheck(t) },
+		ProviderFactories: testAccProviders,
+		Steps: []resource.TestStep{
+			{
+				Config: fmt.Sprintf("%s %s", `
+				provider "argocd" {
+					headers = [
+						"Hello: HiThere",
+					]
+				}`, testAccArgoCDApplicationSimple(name),
+				),
+			},
+		},
+	})
+}
+
 func testAccArgoCDApplicationSimple(name string) string {
 	return fmt.Sprintf(`
 resource "argocd_application" "simple" {

--- a/argocd/resource_argocd_cluster_test.go
+++ b/argocd/resource_argocd_cluster_test.go
@@ -506,6 +506,7 @@ resource "argocd_cluster" "cluster_two" {
 func testAccArgoCDClusterTwiceWithSameServerNoNamesTrimmed(server, server2 string) string {
 	return fmt.Sprintf(`
 resource "argocd_cluster" "cluster_one" {
+  name = "server"
   server = "%s"
   config {
     # Uses Kind's bootstrap token whose ttl is 24 hours after cluster bootstrap.
@@ -516,6 +517,7 @@ resource "argocd_cluster" "cluster_one" {
   }
 }
 resource "argocd_cluster" "cluster_two" {
+  name = "server"
   server = "%s"
   config {
     # Uses Kind's bootstrap token whose ttl is 24 hours after cluster bootstrap.

